src/main/java/org/truffleruby/core/rope/RopeConstants.java

```diff
@@ -126,4 +127,49 @@ public static LeafRope lookupUSASCII(String string) {
         }
     }
 
+    private static final LeafRope[] PADDED_NUMBERS = createPaddedNumbersTable();
+
+    private static LeafRope[] createPaddedNumbersTable() {
+        final LeafRope[] table = new LeafRope[100];
+
+        for (int n = 0; n < table.length; n++) {
+            table[n] = new AsciiOnlyLeafRope(
+                    new byte[]{ (byte) ('0' + n / 10), (byte) ('0' + n % 10) },
+                    UTF8Encoding.INSTANCE);
+        }
+
+        return table;
+    }
+
+    public static LeafRope paddedNumber(int n) {
```

Comment:
Could you add some JavaDoc here to mention it's like %02d for values between 0 and 99?

Reply:
Done.